typescript/packages/plugins/crossmint-headless-checkout/src/crossmint-headless-checkout.plugin.ts

```diff
@@ -0,0 +1,102 @@
+export class CrossmintHeadlessCheckoutPlugin extends PluginBase {
+    private readonly crossmintApiClient: CrossmintApiClient;
+
+    constructor(
+        private readonly crossmint: Crossmint,
+        private readonly callDataSchema: z.ZodSchema,
+    ) {
+        super("crossmint-headless-checkout", []);
+
+        const validatedCrossmint = createCrossmint(crossmint, {
+            usageOrigin: "server",
+        });
+        this.crossmintApiClient = new CrossmintApiClient(validatedCrossmint, {
+            internalConfig: {
+                sdkMetadata: {
+                    name: "@goat-sdk/plugin-crossmint-headless-checkout",
+                    version: packageJson.version,
+                },
+            },
+        });
+    }
+
+    supportsChain(chain: Chain): boolean {
+        return chain.type === "evm"; // TODO: Add support for more blockchains
+    }
+
+    async getTools(walletClient: EVMWalletClient) {
+        const superTools = await super.getTools(walletClient);
+        return [
+            ...superTools,
+            createTool(
+                {
+                    name: "create_and_pay_order",
+                    description: "Create and pay for a Crossmint Headless Checkout order",
+                    parameters: getCreateAndPayOrderParameters(this.callDataSchema),
+                },
+                async (params) => {
+                    const res = await this.crossmintApiClient.post("/api/2022-06-09/orders", {
+                        body: JSON.stringify(params),
+                        headers: {
+                            "x-api-key": this.crossmint.apiKey,
+                            "Content-Type": "application/json",
+                        },
+                    });
+
+                    if (!res.ok) {
+                        // biome-ignore lint/suspicious/noImplicitAnyLet: <explanation>
+                        let json;
+                        try {
+                            json = await res.json();
+                            throw new Error(
+                                `Failed to create order: ${res.status} ${res.statusText}\n\n${JSON.stringify(json, null, 2)}`,
+                            );
+                        } catch (e) {
+                            throw new Error(`Failed to create order: ${res.status} ${res.statusText}`);
+                        }
+                    }
+
+                    const { order } = (await res.json()) as { order: Order; orderClientSecret: string };
+
+                    const serializedTransaction =
+                        order.payment.preparation != null && "serializedTransaction" in order.payment.preparation
+                            ? order.payment.preparation.serializedTransaction
+                            : undefined;
+                    if (!serializedTransaction) {
+                        throw new Error(
+                            `No serialized transaction found for order:\n\n ${JSON.stringify(order, null, 2)}`,
```

Comment:
when would this ever happen? can the error be more specific about how to address it

Reply:
mostly if the item isnt available

Reply:
hmm at a minimum add a TODO for that, we can improve it in a future version